### ISSUE TITLE
[Question] Help with weave net setup

### ISSUE THREAD

**Issue by thomkle** [hobby-kube/guide#83]
Hi,

First of all, let me thank you for this amazing guide. I'm very new to kubernetes and having a guide like this to follow helps a lot when trying to setup my first cluster!

That being said, I'm having some issues with deploying the cert-manager.
It seems to just be stuck on `ContainerCreating`. I did some troubleshooting and got some more info using `kubectl describe pod -n ingress cert-manager`: 

> Failed to create pod sandbox: rpc error: code = Unknown desc = failed to set up sandbox container "deaffba461d577133a8aeccec4928f9c9294de1b4a3ab6907ed0aacf188bc12f" network for pod "cert-manager-6d56fb9594-88l4p": networkPlugin cni failed to set up pod "cert-manager-6d56fb9594-88l4p_ingress" network: netplugin failed with no error message

I have no idea where to go from here. I've googled keywords like "networkPlugin cni failed to set up pod" and "Failed to create pod sandbox: rpc error", but none of the solutions I could find had a solution for my problem. I saw some issues mentioning weave net, might this be the problem? Maybe I set it up wrong, but I'm pretty certain I followed the instructions pretty well. Any help would be greatly appreciated!

**Comment by thomkle** [comment]
So I got it working by running `ufw allow 10250` and `ufw allow 6783`, but I got a feeling that it's not the correct way of doing it. The connections should work through the weave interface and that should already be allowed with the command that's mentioned in the guide `ufw allow in on weave`. Do I need to add more ip routes than the 10.96.x.x range? I see a lot of mentions of 10.32.x.x.

Even though I got the weave network to "work" and got to deploy the tls and the dashboard, the dashboard keeps redirect looping to the front page saying 

> the server could not find the requested resource
> Redirecting to previous state in 3 seconds

In the logs for the dashboard it also mentions the 10.32 ip: 

> 2020/05/21 16:58:00 [2020-05-21T16:58:00Z] Incoming HTTP/1.1 GET /api/v1/login/status request from 10.32.0.1:39446: {}
> 2020/05/21 16:58:00 [2020-05-21T16:58:00Z] Outcoming response to 10.32.0.1:39446 with 200 status code
> 2020/05/21 16:58:00 [2020-05-21T16:58:00Z] Incoming HTTP/1.1 GET /api/v1/overview?filterBy=&itemsPerPage=10&name=&page=1&sortBy=d,creationTimestamp request from 10.32.0.1:39446: {}
> 2020/05/21 16:58:00 Getting config category
> 2020/05/21 16:58:00 Getting discovery and load balancing category
> 2020/05/21 16:58:00 Getting lists of all workloads
> 2020/05/21 16:58:00 the server could not find the requested resource
> 2020/05/21 16:58:00 [2020-05-21T16:58:00Z] Outcoming response to 10.32.0.1:39446 with 404 status code

**Comment by godwhoa** [comment]
I remember running into issues with weave net when I tried Hobby Kube more than a year ago. 
I have this bookmarked from back then, maybe it's the same issue you are running into: 
https://github.com/weaveworks/weave/issues/2736

**Comment by pstadler** [comment]
Not sure what the problem is, but my first guess would be a missing kernel module or parameter. Where are you hosting this?

**Comment by pstadler** [comment]
Actually, could some of the notes from https://github.com/hobby-kube/guide/issues/74 fix this?

**Comment by thomkle** [comment]
@godwhoa Thanks for the input, but if network-overlaping was the issue, opening the ports that I did to make the weave-part work would probably not have solved the weave-part of the issue? I might be off on that.

@pstadler I'm hosting it using 3 vpns from Hetzner. I didn't use their new private network. So I did the 4. step and that made it possible for me to remove the two ports that I needed to allow to make weave work! So, that part looks to be solved, thanks! However, the dashboard still seems to not work properly, but navigating directly to some of the sub-menu items (like /#!/node) works and stops the redirect-loop. Some of the sub-menu items triggers the redirection (404) again (like /#!/replicaset). This might even be an issue within the dashboard itself and not related to the setup, I find it hard to tell since I've never done anything with Kubernetes before following this guide.
